### PR TITLE
Implement design refinements to Material Table in some random places

### PR DESCRIPTION
**Pull request by tillyw**
## Associated issues
https://github.com/cityofaustin/atd-data-tech/issues/7139

## Testing
**URL to test:** 
https://deploy-preview-532--atd-moped-main.netlify.app/moped/projects

**Steps to test:**
Open different tables and check formatting

---
#### Ship list
- [x] Code reviewed 
- [x] Product manager approved
- [x] ~Added to [QA test script](https://docs.google.com/spreadsheets/d/1n_O6MLh9cwwPf57HUM394Ea-z9uuoEV1-QW4axNZXLE/edit#gid=776973707)~

### CODE REVIEW

File: moped-editor/src/views/projects/projectView/ProjectFundingTable.js [hunk 1]
```diff
@@ -547,7 +552,11 @@ const ProjectFundingTable = () => {
         data={data.moped_proj_funding}
         title={
           <div>
-            <Typography variant="h2" color="primary">
+            <Typography
+              variant="h2"
+              color="primary"
+              style={{ paddingTop: "1em", paddingBottom: "1em" }}
```
Comment (mateoclarke) [inline moped-editor/src/views/projects/projectView/ProjectFundingTable.js:558]: I think only the `paddingTop` is needed, no extra `paddingBottom` unless the [mock](https://user-images.githubusercontent.com/1463708/150427836-a4e83698-cff0-470d-8512-14ea79a1b6b1.png) changed

**Review by mateoclarke — APPROVED** [review]
I think the `min-width: 8em` is still missing from Files - type field

Other than that and the extra padding, looks all good! 🚢 💨 ✨

**Comment by amenity** [comment]
@tillyw - thanks! Looking so nice! ✨ 

I think there is now a _max_ width being applied to  "Name" field in the "Team" table. Unfortunately, that makes the dropdown much harder to use because the names wrap to second lines. 
9tNNA

[Same goes for the "Milestone name" field](https://user-images.githubusercontent.com/1463708/151897653-dd39ad53-a5e1-4474-9858-a480e92554ca.png).

@mateoclarke is right about the `min-width` still missing from Files - type field... But that was a mistake in my original issue: That field only presents as empty to users in a modal and already renders wide enough. 

So, that leaves these small tweaks:
- [ ] For the "Project team" "Name" field, please just leave as is width-wise and show the "Required" help text. 
- [ ] For the "Milestone name" "Milestone name" field, please just leave as is width-wise and show the "Required" help text. 
- [ ] [Remove extra bottom padding](https://github.com/cityofaustin/atd-moped/pull/532#discussion_r794969155)

**Review by amenity — APPROVED** [review]
Thank you, @tillyw ! Looks so sharp. 🙏🏻 ✨

**Review by mateoclarke — APPROVED** [review]
🚢💨✨

**Pull request merged**